src/coreclr/jit/codegenarm64.cpp

```diff
@@ -4209,26 +4209,55 @@ void CodeGen::genCodeForStoreInd(GenTreeStoreInd* tree)
         {
             bool addrIsInReg   = addr->isUsedFromReg();
             bool addrIsAligned = ((tree->gtFlags & GTF_IND_UNALIGNED) == 0);
+            bool handledByRcpc = false;
 
-            if ((ins == INS_strb) && addrIsInReg)
+            // On arm64-v8.4+ we can use stlur* instructions with acquire/release semantics
+            // if the address is LEA with just imm offset (unscaled)
+            if (compiler->compOpportunisticallyDependsOn(InstructionSet_Rcpc2) && addrIsAligned &&
+                (tree->Addr()->OperIs(GT_LEA) && !tree->HasIndex() && (tree->Scale() == 1) &&
+                 emitter::emitIns_valid_imm_for_unscaled_ldst_offset(tree->Offset())))
```

Comment:
Looks like an unnecessary extra level of parens here.
```suggestion
                tree->Addr()->OperIs(GT_LEA) && !tree->HasIndex() && (tree->Scale() == 1) &&
                emitter::emitIns_valid_imm_for_unscaled_ldst_offset(tree->Offset()))
```